xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "InputStreamAddon.h"
+#include "addons/InputStream.h"
+#include "cores/VideoPlayer/DVDClock.h"
+
+CInputStreamAddon::CInputStreamAddon(CFileItem& fileitem, ADDON::CInputStream *inputStream)
+: CDVDInputStream(DVDSTREAM_TYPE_ADDON, fileitem), m_addon(inputStream)
+{
+  m_hasDemux = false;
+}
+
+CInputStreamAddon::~CInputStreamAddon()
+{
+  Close();
+  m_addon->Stop();
+  delete m_addon;
+}
+
+bool CInputStreamAddon::Open()
+{
+  bool ret = false;
+  if (m_addon)
+    ret = m_addon->Open(m_item);
+  if (ret)
+  {
+    m_hasDemux = m_addon->HasDemux();
+    m_hasDisplayTime = m_addon->HasDisplayTime();
+    m_hasPosTime = m_addon->HasPosTime();
+    m_canPause = m_addon->CanPause();
+    m_canSeek = m_addon->CanSeek();
+  }
+  return ret;
+}
+
+void CInputStreamAddon::Close()
+{
+  if (m_addon)
+    return m_addon->Close();
+}
+
+bool CInputStreamAddon::IsEOF()
+{
+  return false;
+}
+
+int CInputStreamAddon::Read(uint8_t* buf, int buf_size)
+{
+  if (!m_addon)
+    return -1;
+
+  m_addon->ReadStream(buf, buf_size);
```

Comment:
I should say thank *you*. this PR was perfectly timed, I just started `CInputStreamGame` and guess who's work I'm borrowing from ;)